replication/binlogsyncer.go

```diff
@@ -77,6 +78,10 @@ type BinlogSyncer struct {
 
 	nextPos Position
 
+	useGTID bool
```

Comment:
seem you don't need useGTID here, you can use gset == nil to check whether syncs with gtid or not directly.